lintly/parsers.py

```diff
@@ -330,6 +329,56 @@ def parse_violations(self, output):
 
         return violations
 
+class GitLeaksParser(BaseLintParser):
+    """
+    Gitleaks JSON format
+
+        {
+            "line": "-----BEGIN PRIVATE KEY-----",
+            "lineNumber": 59,
+            "offender": "-----BEGIN PRIVATE KEY-----",
+            "commit": "111111111111111111111000000000",
+            "repo": ".",
+            "repoURL": "",
+            "leakURL": "",
+            "rule": "Asymmetric Private Key",
+            "commitMessage": "any commit message \n",
+            "author": "bob s",
+            "email": "bob@example.com",
+            "file": "relative/path/to/output",
+            "date": "2020-04-14T15:17:53-07:00",
+            "tags": "key, AsymmetricPrivateKey"
+            }
+        
+    """
+
+    def parse_violations(self, output):
+        """
+         Parse violations for whatever fuck you like
+        """        
+        if not output:
+            return dict()
+        
+        json_data = output.split("\n")
+        violations = collections.defaultdict(list)
+        for violation_json in json_data:
+            try:
+                violation_data = json.loads(violation_json)
+            except Exception as e:
+                print("Exception parsing json")
+            violation = Violation(
+                line=violation_data["lineNumber"],
+                column=0,
+                code='{}'.format(violation_data["offender"]),
+                #code='{} ({})'.format(violation_data["rule"],violation_data["offender"]),
```

Comment:
Any reason not to do:

```
code=violation_data["offender"]
```
OR
```
code="{violation_data["offender"]} - Tags: {violation_data["tags"]}"
```

1) We don't need the commented out code on line 373.
2) The `.format()` syntax isn't needed if we're doing python3
3) No need to use `.format()` if we're not combining multiple strings.

Reply:
I was little confused here while creating code line.Should we use both rule and offender here or just offender. But yes  with just offender this will also work 
`code=violation_data["offender"]
`